spark2.4/src/main/scala/KafkaReader.scala

```diff
@@ -102,11 +113,17 @@ object KafkaReader {
       reader.option("kafka.client.id", s"$group-$clientId")  // probably should use uuid instead of user input TODO
     }
     
-    val values = reader
-      .load.select(from_json(col("value") cast "string", schema, Map( "timestampFormat" -> "yyyy/MM/dd HH:mm:ss" )) as "data", col("timestamp") cast "long" as "TM_EXT_KAFKA")
-      .select("data.*", "TM_EXT_KAFKA")
-      .withColumn("TM_EXT_KAFKA", col("TM_EXT_KAFKA") * 1000)
-      .withColumn("TM_SSDS", unix_timestamp * 1000 )
+    val values = if( useFlowMarkers ) {
+      reader
+        .load.select(from_json(col("value") cast "string", schema, Map( "timestampFormat" -> "yyyy/MM/dd HH:mm:ss" )) as "data", col("timestamp") cast "long" as "TM_EXT_KAFKA")
+        .select("data.*", "TM_EXT_KAFKA")
+        .withColumn("TM_EXT_KAFKA", col("TM_EXT_KAFKA") * 1000)
+        .withColumn("TM_SSDS", unix_timestamp * 1000 )
+    } else {
+      reader
+        .load.select(from_json(col("value") cast "string", schema, Map( "timestampFormat" -> "yyyy/MM/dd HH:mm:ss" )) as "data")
+        .select("data.*")
```

Comment:
Again, I don't understand why to `from_json` and `Map(...) as "data"` for `select("data.*")`? I seem to be failing to find something important here.

Reply:
I believe this is the same issue as two comments above.  If there's something else, please let me know.